src/math_llama.jl

```diff
@@ -0,0 +1,44 @@
+"""
+    math_llama.jl
+
+    This file contains the math functions used in the Llama model.
+"""
+
+begin
+# Calculate the root mean square norm of a vector. Reference in llama2.c lines 182-195
+function rmsnorm(x::Vector{Float32}, weight::Vector{Float32})
+    size = length(x)
+    # Calculate the sum of the squares
+    sum_squares = sum(x .^ 2) / size
+    sum_squares += 1e-5
```

Comment:
Would probably be better to stay in the Float32 domain, i.e. `1f-5` instead of `1e-5` and `1.0f0` instead of `1.0`. As the rmsnorm function is called pretty often, this might make a small difference :D